### DISCUSSION TITLE
Welcome to MackieC4_P3 Discussions!

### DISCUSSION THREAD

**Discussion by markusschloesser** [markusschloesser/MackieC4_P3#5] [General]
Hi,
this is the discussion board for the repository, let's share ideas here on how to best drive things forward :-)
Be friendly!
Best
Markus

**Comment by BilldarBagdar** [comment]
Hi Markus!

I accepted your invitation, but I am still behind in terms of Live 11.  I will clone this repo and attempt to integrate it into my existing IDE setup.  As you said the difference between 2.7 and 3 shouldn't really impact the script code per se.  I will stay tuned.

**Reply by markusschloesser** [reply]
Great!
Thibaultlebrun  suggested to get started using decorators for the whole "listener" stuff and I am in the middle of familiarizing myself with those. When I tried integrating one yesterday, was the first time that pycharm complained about non compatibility with py2.7.

**Reply by BilldarBagdar** [reply]
I cloned the repo and opened it in my PyCharm environment without issue.  I can still navigate around by Ctrl + Alt + b (US keyboard) which is huge.  Do you know about Ctrl+Alt+b?  If you are reading code and see a method call and you want to know what that method does, you can highlight the method name and Ctrl+Alt+b will take you straight to the definition of that method wherever it is. (Of course, this only works if you have decompiled versions of all the classes and methods you would want to Ctrl+Alt+b into referenced so your IDE knows about them.) The other side of that coin I use a lot is if you are reading code in some method and you want to know where the method is called, you can highlight the method name, right-click and select "Find Usages..."

**Reply by markusschloesser** [reply]
I use "kite" as a plugin in pycharm, which brings the definition via hover (and more), can only recommend it!
